### PR TITLE
Add function for logging function calls

### PR DESCRIPTION
**Pull request by xylar**
The function `log_function_call()` is similar to `log_method_call()` but for functions instead of methods.

This will be useful in future work where we are changing some methods to be functions instead.

### CODE REVIEW

**Comment by xylar** [comment]
## Testing

I ran the ocean `nightly` test suite on my Ubuntu laptop.  All tests ran successfully and I see:
```
compass calling: compass.parallel.set_cores_per_node()
  in /home/xylar/code/compass/generalize_log_method_call/compass/parallel.py
```
as expected in the test-case log file when I added a test call to `log_function_call()`.

File: compass/logging.py [hunk 1]
```diff
@@ -64,3 +63,29 @@ def log_method_call(method, logger):
...
+      compass calling: compass.parallel.set_cores_per_node()
+        in /home/xylar/code/compass/compass/compass/parallel.py
+
+    Parameters
+    ----------
+    function : function
+        The function that will be run immediately following this call
+
+    logger: logging.Logger
+        The logger to log the method path and file path to
```
Comment (altheaden) [inline compass/logging.py:81]: This should be `The logger to log the function path and file path to`

**Review by altheaden — APPROVED** [review]
Other than the one typo, this looks good to me.

File: compass/logging.py (hunk 1 above)
Comment (xylar) [reply compass/logging.py]: Done!

**Comment by xylar** [comment]
Thanks @altheaden.  Please give it another look and try testing this branch as well if you get a chance.

**Comment by xylar** [comment]
@altheaden, I'm going to merge this based on my testing.  We can always make further fixes if needed.

**Pull request merged**